internal/pkg/coordinator/monitor.go

```diff
@@ -188,10 +189,29 @@ func (m *monitorT) handlePolicies(ctx context.Context, hits []es.HitT) error {
 	return nil
 }
 
+func isAvailabilityError(e error) bool {
+	if e == nil {
+		return false
+	}
+	match, err := regexp.MatchString("connect: connection refused$|net/http: timeout awaiting response headers$", e.Error())
+	if err != nil {
+		log.Warn().Err(err).Msg("Error occured while parsing error message")
+		return false
+	}
+	return match
+}
+
 // ensureLeadership ensures leadership is held or needs to be taken over.
 func (m *monitorT) ensureLeadership(ctx context.Context) error {
 	m.log.Debug().Msg("ensuring leadership of policies")
 	err := dl.EnsureServer(ctx, m.bulker, m.version, m.agentMetadata, m.hostMetadata, dl.WithIndexName(m.serversIndex))
+
+	// If we are doing a read on an unavailable ES instance ignore it and we will check back later.
+	if isAvailabilityError(err) {
+		log.Info().Err(err).Msgf("Encountered availability error while attempting ES read; continuing to retry.")
```

Comment:
Could we also log the hostname we tried to connect to?

Lets used Elasticsearch and not ES as an abbreviation as this is user facing.